### DISCUSSION TITLE
--resource-path and \graphicspath

### DISCUSSION THREAD

**Discussion by bgrenet** [jgm/pandoc#9360] [Q&A] [answered]
As far as I can tell, --resource-path is not used when converting a markdown file to .tex, standalone. Consider the following input file `mwe.md`:

`This is an image : ![](img.jpg)`

- If `img.jpg` is in the same directory as `mwe.md`, one can do either:
  - `pandoc mwe.md -o mwe.pdf` 
  - `pandoc mwe.md -s -o mwe.tex` and then `pdflatex mwe.tex`
 

- Now if `img.jpg` is in a subdirectory `fig/`, one can do `pandoc mwe.md --resource-path=fig/ -o mwe.pdf`. But it is not possible to go through .tex since `pandoc mwe.md --resource-path=fig/ -s -o mwe.tex` will produce the exact same file as without the option `--resource-path`. A workaround would be that `--resource-path` makes the LaTeX writer issue a `\graphicspath{{fig/}}`. Another solution would be for pandoc to look for img.jpg, and directly issue an `\includegraphics{fig/img.jpg}`. 

Does at least one solution make sense?

P.S.: The exact same issue occurs if for images in SVG, with `\includesvg` and `\svgpath` from the package `svg`.

**Comment by jgm** [comment]
`--resource-path` is only used when pandoc itself tries to find images.  When it's generating TeX, it does not need to, so `--resource-path` does nothing.

We could think about the two options you gave, but I'm a bit worried about expanding expectations around `--resource-path`.  (E.g., if we have the TeX writer try to find images using the resource path and rewrite image paths, shouldn't we do that for *all* the writers?)

**Reply by bgrenet** [reply]
Maybe my real request is a way to access the .tex file produced by pandoc when creating a pdf file through LaTeX. This would be sometimes useful when some compilation errors occur during the call to pandoc. More generally, having a way to easily and faithfully produce a .tex file that can produce the same output as `pandoc input.md -o output.pdf` would be a great for all the occasions where one is asked to produce a .tex file that will be compiled (say arXiv, some editors, …).

I would have thought that `--resource-path=...` + `--standalone` is the way to go, but if there is another one it is as good. (Maybe an option `--keep-tex-file` or something like that?)

**Reply by jgm** [reply, marked as answer]
One option here is to use `--pdf-engine=latexmk --pdf-engine-opt=-outdir=build`. Then you can look in the (persistent) `build` directory and find the tex file.

Another option is simply to use `--verbose` and capture stderr; the entire tex file will be there.

**Reply by cagix** [reply]
> `--resource-path` is only used when pandoc itself tries to find images.  When it's generating TeX, it does not need to, so `--resource-path` does nothing.

When reading the current [documentation](https://pandoc.org/MANUAL.html#general-writer-options) this was not clear to me. Since this option appears in section “general writer options”, it would help if you could add this sentence to the documentation, please?

Edit: Opened https://github.com/jgm/pandoc/pull/9417 to clarify the relevant section in the man page.
